apps/backend/src/automation/job.ts

```diff
@@ -0,0 +1,45 @@
+import { invariant } from "@argos/util/invariant";
+
+import { AutomationActionRun } from "@/database/models/index.js";
+import { createModelJob, UnretryableError } from "@/job-core/index.js";
+
+import { getAutomationAction } from "./actions";
+import { AutomationActionContext } from "./defineAutomationAction";
+
+async function processAutomationActionRun(
+  automationActionRun: AutomationActionRun,
+): Promise<void> {
+  // Get the action
+  const actionDefinition = getAutomationAction(automationActionRun.action);
+
+  invariant(
+    actionDefinition,
+    `Unsupported action type: ${automationActionRun.action}. Action not found in registry.`,
+    UnretryableError,
+  );
+
+  // Validate the payload schema
+  const {
+    success: parsingSuccess,
+    error: parsingError,
+    data: validatedPayload,
+  } = actionDefinition.payloadSchema.safeParse(
+    automationActionRun.actionPayload,
+  );
+
+  invariant(
+    parsingSuccess,
+    `Invalid payload for action ${automationActionRun.action}: ${parsingError}`,
+    UnretryableError,
+  );
+
+  // Process the action
+  const jobContext: AutomationActionContext = { automationActionRun };
+  await actionDefinition.process({ ...validatedPayload, ctx: jobContext });
```

Comment:
Wrap in a try / catch, if it's an automation error, patch it and at the end if no error patch it completed with the timestamps. Relative to my other comments